### PR TITLE
doc: add documentation for board configurator

### PR DESCRIPTION
**Pull request by greg-fer**
Added documentation framework and the basic user guides for the app. NCD-640.

----

- [x] Update images
- [ ] Fix prettifier

### CODE REVIEW

File: doc/docs/overview.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Select Device
+
+Opens the drop-down with the list of devices connected to the computer. You can choose a device from the list of connected devices to perform further actions on the device such as programming.
+
+## Actions
+
+When you select a device, the following actions become available in the side panel:
+
+- **Write config** - Writes the selected configurations to the development kit.
```
Comment (greg-fer) [inline doc/docs/overview.md:17]: @cybic , is this the only action we get to use in this version? If so, are you going to hide the placeholders for other actions?

File: doc/docs/overview.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Overview and user interface
+
+After starting Board Configurator, the main application window is displayed.
+
+![Board Configurator application window](./screenshots/board_ctrl_overview_alt.png "Board Configurator application window")
```
Comment (greg-fer) [inline doc/docs/overview.md:5]: Note to self: Needs update.

File: doc/docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+When you select a device, the following actions become available in the side panel:
+
+- **Write config** - Writes the selected configurations to the development kit.
+
+## Configuration tab
+
+In the **Configuration** tab, you can see the options that are configurable for the selected development kit.
+
+![Board Configurator configuration tab](./screenshots/board_ctrl_connected.png "Board Configurator configuration tab")
```
Comment (greg-fer) [inline doc/docs/overview.md:23]: Note to self: Needs update.

**Review by chunfantasy — COMMENTED** [review]
The screenshots will be outdated soon when the name is changed from Board Controller to Board Configurator.

File: doc/docs/overview.md (hunk 1 above)
Comment (cybic) [reply doc/docs/overview.md:17]: The placeholders have already been hidden, so yes!

**Comment by greg-fer** [comment]
> The screenshots will be outdated soon when the name is changed from Board Controller to Board Configurator.

Updated with new screenshots from @cybic .

**Comment by greg-fer** [comment]
@cybic , @chunfantasy , can you help me solve the Azure check failing? It's the prettifier complaining about my markdown. I want to in turn complain about the prettifier. It's its word against mine...

**Review by chunfantasy — APPROVED** [review]
LGTM

**Review by aadnekar — APPROVED** [review]

**Comment by greg-fer** [comment]
Updated revision history and the images. @cybic , kindly please help with prettifier again, review, and approve if everything looks ok.

File: doc/docs/overview.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+After starting Board Configurator, the main application window is displayed.
+
+![Board Configurator application window](./screenshots/board_configurator_overview.png "Board Configurator application window")
+
+The available options and information change after you **Select Device**.
+
+## Select Device
+
+Opens the drop-down with the list of devices connected to the computer. You can choose a device from the list of connected devices to perform further actions on the device such as programming.
```
Comment (cybic) [inline doc/docs/overview.md:11]: ".. such as configuring"?

**Review by cybic — APPROVED** [review]
One small comment. LGTM!

**Pull request merged**